server/redeemer.go

```diff
@@ -0,0 +1,378 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/big"
+	gonet "net"
+	"net/url"
+	"sync"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/net"
+	"github.com/livepeer/go-livepeer/pm"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var cleanupLoopTime = 5 * time.Minute
+
+type localSenderMonitor interface {
+	pm.SenderMonitor
+	SubscribeMaxFloatChange(sender ethcommon.Address, sink chan<- struct{}) event.Subscription
+}
+
+type Redeemer struct {
+	server    *grpc.Server
+	recipient ethcommon.Address
+	eth       eth.LivepeerEthClient
+	sm        localSenderMonitor
+	quit      chan struct{}
+}
+
+// NewRedeemer creates a new ticket redemption service instance
+func NewRedeemer(recipient ethcommon.Address, eth eth.LivepeerEthClient, sm *pm.LocalSenderMonitor) (*Redeemer, error) {
+
+	if recipient == (ethcommon.Address{}) {
+		return nil, fmt.Errorf("must provide a recipient")
+	}
+
+	if eth == nil {
+		return nil, fmt.Errorf("must provide a LivepeerEthClient")
+	}
+
+	if sm == nil {
+		return nil, fmt.Errorf("must provide a SenderMonitor")
+	}
+
+	return &Redeemer{
+		recipient: recipient,
+		eth:       eth,
+		sm:        sm,
+		quit:      make(chan struct{}),
+	}, nil
+}
+
+// Start starts a Redeemer server
+func (r *Redeemer) Start(host *url.URL) error {
+	listener, err := gonet.Listen("tcp", host.String())
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+	if err != nil {
+		return err
+	}
+
+	s := grpc.NewServer()
+	r.server = s
+
+	net.RegisterTicketRedeemerServer(s, r)
+
+	return s.Serve(listener)
```

Comment:
That's right it will block. Currently however the caller of `Start` wraps this in a goroutine. Want to change this to have the callee start the goroutine? 

https://github.com/livepeer/go-livepeer/blob/c63b8c0afc3afca9ec1f2f49a836d6f9ec335eb4/cmd/livepeer/livepeer.go#L619

Reply:
Got it. I'm cool with having the caller of `Start()` wrap it in a goroutine. I think it would be helpful to add a comment on this method noting that it will block.